posts/atomic-design-in-practice.md

```diff
@@ -0,0 +1,233 @@
+
+```scss
+.organism-header {
+  display: flex;
+  > [class^="organism-header"] {
+    margin-right: 40px;
+    align-self: flex-end;
+    &:last-child {
+      margin-right: 0;
+    }
+  }
+  .organism-header-navigation {
+    flex-grow: 1;
+  }
+  .organism-header-form-search {
+    width: 250px;
+  }
+}
+```
+
+# Templates and pages
+
+![The template on the left side and the page, the instance, on the right side](https://raw.githubusercontent.com/Gnuk/blog-usa/atomic-design-in-practice/images/2019/10/atomic-templates-pages.png)
+
+Templates represents the page structure. This step is just a canvas to put our organisms. The template will be the model of our page, so the page is an instance of a template.
+
+![Template and page example](https://raw.githubusercontent.com/Gnuk/blog-usa/atomic-design-in-practice/images/2019/10/atomic-templates-pages-example.png)
+_Template and page inspired from the Brad Frost [article](http://bradfrost.com/blog/post/atomic-web-design)_
+
+# Organisation
+
+From previous examples, we can notice the decoupling of structures (HTML) and styles (CSS). These parts can be organised in many separate files using template languages like [Pug](https://pugjs.org) and [Sass](https://sass-lang.com/).
+
+For an organisational structure, I've the same example stored in several files on GitLab: https://gitlab.ippon.fr/arey/atomicdesign
+
+# As a conclusion and to go further
+
+As we have seen, making Atomic Design components require to cut elements. These elements have their responsibilities (label atom, form molecule …), they can be modified independently. Moreover, their assembly is done in a structured way (atoms, molecules, organisms): they are consistent and maintainable.
+
+Decoupling can go even further by trying to use your own components instead of a specific framework.
+
+On Angular and Vue, I encourage you to make your style outside of your components using Sass in an external library, because, if you have a strong coupling, you may meet the following problems:
+
+* Global tokens (colors, font size, font family …) are less accessible and more complex to maintain.
+* Component assembly (molecules, organisms) is less consistent and more difficult to manage.
+* Global style vision is more complex because of components containing a lot of responsibilities: HTML template, TypeScript or JavaScript logic and CSS style.
+
+Sass usage to make a single file with several parts from atoms, molecules and organisms have the following advantages:
+
+* Management of token [variables](https://sass-lang.com/guide#topic-2)
+* Molecules and organisms management and definition using [imports](https://sass-lang.com/guide#topic-5)
+* Unique style distribution from a single CSS file that can be imported in your HTML using `<link>` tag
+
+> This separation is not mandatory, but, I strongly encourage you to do it, because, in my experience, I had a lot of similar Angular or Vue components, with domain responsibilities but a common appearance.
```

Comment:
This separation is not mandatory but I strongly encourage you to do it because in my experience, I had a lot of similar Angular or Vue components with domain responsibilities but a common appearance.